### ISSUE TITLE
findutils/xargs has a subtle undeclared runtime dependency on an external echo command

### ISSUE THREAD

**Issue by abathur** [NixOS/nixpkgs#89055]
**Describe the bug**
If you run `xargs` without specifying the command argument, it falls back on running `echo` (this is roughly documented in the man page, but the note there mistakenly claims the default is `/bin/echo`, but this was changed back in 2011...). If it can't find echo, xargs throws a `.../bin/xargs: echo: No such file or directory` error.

**To Reproduce**
This example is a little synthetic (not very similar to how I found the issue), but a simple way to demonstrate it is: `nix-shell -p findutils --run 'echo heh | env -i PATH= $(type -p xargs)'`

**Additional context**
The checkPhase of the CI build for my shell-script dependency resolver invokes a test script like `env -i <script>` to demonstrate that it's depending on resolved executable paths--not the environment. The test script in turn sources a resolved copy of shunit2 (which contains a bare `xargs` invocation). This has been running okay in macOS and Linux (but obviously, with a hidden dependency leaking in), but I noticed today that it errors on NixOS.

It's not hard to patch this with substitute:
```
find_some_utils = pkgs.findutils.overrideAttrs (oldAttrs: rec {
    prePatch = ''
      substituteInPlace xargs/xargs.c --replace 'char default_cmd[] = "echo";' 'char default_cmd[] = "${coreutils}/bin/echo";'
    '';
  });
```

I'm not, however, sure if it is "right" to try to fix this at all, nor if this is the right approach.

**Notify maintainers**
@veprbl @disassembler @vcunat

**Comment by veprbl** [comment]
@abathur Open a PR for this?

**Comment by abathur** [comment]
@veprbl Sure. Still counting on someone else to make sure this is appropriate, though.

**Comment by abathur** [comment]
This was fixed by #89055
